python/wavefrontsender.md

```diff
@@ -0,0 +1,133 @@
+
+The `WavefrontSender` interface has two implementations:
+
+* Option 1: [Create a `WavefrontProxyClient`](#Option-1-Sending-Data-via-the-Wavefront-Proxy) to send data to a Wavefront proxy.
+* Option 2: [Create a `WavefrontDirectIngestionClient`](#Option-2-Sending-Data-via-Direct-ingestion) to send data directly to a Wavefront service.
+
+## Option 1: Sending Data via the Wavefront Proxy
+
+>**Prerequisite**
+>Before your application can use a `WavefrontProxyClient`, you must [set up and start a Wavefront proxy](https://docs.wavefront.com/proxies_installing.html).
+
+When sending data via the Wavefront proxy, you need to create a `WavefrontProxyClient`. Include the following information.
+
+* The name of the host that will run the Wavefront proxy.
+* One or more proxy listening ports to send data to. The ports you specify depend on the kinds of data you want to send (metrics, histograms, and/or trace data). You must specify at least one listener port. 
+* Optional settings for tuning communication with the proxy.
+
+> **Note**: See [Advanced Proxy Configuration and Installation](https://docs.wavefront.com/proxies_configuring.html) for details:
+
+```python
+from wavefront_sdk import WavefrontProxyClient
+
+# Create a sender with:
+   # the proxy hostname or address
+   # the recommended listener port (30000) for sending trace data 
+   # the recommended listener port (2878) for sending histograms
+   # the default listener port (2878) for sending metrics  
+   # a nondefault interval (2 seconds) for flushing data from the sender to the proxy. Default: 5 seconds
+wavefront_sender = WavefrontProxyClient(
+   host="<PROXY_HOST>",
+   tracing_port=30000,
+   distribution_port=2878,
+   metrics_port=2878,
+   internal_flush=2
+)
+```
+ 
+> **Note:** When you set up a Wavefront proxy on the specified proxy host, you specify the port it will listen to for each type of data to be sent. The `WavefrontProxyClient` must send data to the same ports that the Wavefront proxy listens to. Consequently, the port-related builder methods must specify the same port numbers as the corresponding proxy configuration properties. See the table given below:  
+
+| `WavefrontProxyClient()` parameter | Corresponding property in `wavefront.conf` |
+| ----- | -------- |
+| `metrics_port` | `pushListenerPorts=` |
+| `distribution_port` | `histogramDistListenerPorts=` |
+| `tracing_port` | `traceListenerPorts=` |
+
+## Option 2: Sending Data via Direct ingestion
+
+When sending data via direct ingestion, you need to create a `WavefrontDirectIngestionClient`, and build it with the Wavefront URL and API token to send data directly to Wavefront.
+
+>**Prerequisite**
+> * Verify that you have the Direct Data Ingestion permission. For details, see [Examine Groups and Permissions](https://docs.wavefront.com/users_account_managing.html#examine-groups-and-permissions).
```

Comment:
This link is changing to Examine Groups, Roles, and Permissions (on my sandbox at http://docs-sandbox-a.wavefront.com/users_account_managing.html#examine-groups-roles-and-permissions . But, only when 56 goes out.

Reply:
Ack! Created a JIRA to track this.